src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -100,48 +196,69 @@
+  [event initial-y scroll-y]
+  (let [content-size-y (- (oops/oget event "nativeEvent.contentSize.height")
+                          (oops/oget event "nativeEvent.layoutMeasurement.height"))
+        current-y      (+ (oops/oget event "nativeEvent.contentOffset.y") initial-y)]
+    (reanimated/set-shared-value scroll-y (- content-size-y current-y))))
 
 (defn messages-list-content
-  [{:keys [chat-id] :as chat} insets keyboard-shown]
-  (fn []
-    (let [context  (rf/sub [:chats/current-chat-message-list-view-context])
-          messages (rf/sub [:chats/raw-chat-messages-stream chat-id])]
-      [rn/view
-       {:style {:flex 1}}
-       ;; NOTE: DO NOT use anonymous functions for handlers
-       [rn/flat-list
-        {:key-fn                       list-key-fn
-         :ref                          list-ref
-         :header                       [list-header chat]
-         :footer                       [list-footer chat]
-         :data                         messages
-         :render-data                  {:context        context
-                                        :keyboard-shown keyboard-shown}
-         :render-fn                    render-fn
-         :on-viewable-items-changed    on-viewable-items-changed
-         :on-end-reached               list-on-end-reached
-         :on-scroll-to-index-failed    identity ; don't remove this
-         :content-container-style      {:padding-top    (+ composer.constants/composer-default-height
-                                                           (:bottom insets)
-                                                           32)
-                                        :padding-bottom 16}
-         :scroll-indicator-insets      {:top (+ composer.constants/composer-default-height
-                                                (:bottom insets))}
-         :keyboard-dismiss-mode        :interactive
-         :keyboard-should-persist-taps :handled
-         :on-momentum-scroll-begin     state/start-scrolling
-         :on-momentum-scroll-end       state/stop-scrolling
-         :scroll-event-throttle        16
-         :on-scroll                    on-scroll
-         ;; TODO https://github.com/facebook/react-native/issues/30034
-         :inverted                     (when platform/ios? true)
-         :style                        (when platform/android? {:scaleY -1})
-         :on-layout                    on-messages-view-layout}]])))
-
-;; This should be replaced with keyboard hook. It has to do with flat-list probably. The keyboard-shown
-;; value updates in the parent component, but does not get passed to the children.
-;; When using listeners and resetting the value on an atom it works.
+  [{:keys [chat insets initial-y scroll-y cover-bg-color keyboard-shown?]}]
+  (let [context  (rf/sub [:chats/current-chat-message-list-view-context])
+        messages (rf/sub [:chats/raw-chat-messages-stream (:chat-id chat)])]
+    [rn/view {:style {:flex 1}}
+     [reanimated/flat-list
```

Comment:
As I mentioned earlier, this is not a reanimated flat-list. So if normal flat list is sufficient you can change it to `rn/flat-list`.